python/lsst/meas/extensions/scarlet/io/model_data.py

```diff
@@ -0,0 +1,156 @@
+# This file is part of meas_extensions_scarlet.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+from typing import Any
+
+import numpy as np
+from numpy.typing import DTypeLike
+
+import lsst.scarlet.lite as scl
+
+from .source_data import IsolatedSourceData
+
+CURRENT_SCHEMA = "1.0.0"
+SCARLET_LITE_SCHEMA = "1.0.0"
+MODEL_TYPE = "lsst"
+scl.io.migration.MigrationRegistry.set_current(MODEL_TYPE, CURRENT_SCHEMA)
+
+# Ensure that the ScarletModelData from scarlet lite hasn't changed.
+if scl.io.model_data.CURRENT_SCHEMA != SCARLET_LITE_SCHEMA:
+    raise RuntimeError(
+        "Version mismatch between meas_extensions_scarlet and scarlet lite. "
+        "This requires updating SCARLET_LITE_SCHEMA, CURRENT_SCHEMA, and a migration step "
+        f"to match the ScarletModelData schema version {scl.io.model_data.CURRENT_SCHEMA}."
+    )
+
+
+class LsstModelData(scl.io.ScarletModelData):
+    """A ScarletModelData that includes isolated sources.
+
+    Attributes
+    ----------
+    isolated :
```

Comment:
Does this style with the types omitted work?

(I'm hoping it will when we have a new Sphinx that can read type annotations, but I don't think we're there yet.)

Reply:
Sigh. No, this is just a habit from my other projects because Sphinx does automatically use type annotations if they're available, and apparently so does documenteer, but not the versions that we're using. ☹️